### ISSUE TITLE
The goal shere and the robot end-effector do not completely align

### ISSUE THREAD

**Issue by Erfi** [shamilmamedov/flexible_arm#15]
***Description**:
MPC can control the arm and bring it to the goal, but the robot arm is always slightly misaligned with the goal sphere. The misalignment seems consistent in direction: as if the end-effector is slightly longer. 

**Suggestions**:
No idea yet, but the coordinates of the goal set by the env and received by the controller are the same. Perhaps it is a URDF thing?
Is the sphere's coordinate at the center?

**Acceptance Criteria**:
Run the data collection script (```tests/test_mpc_data_collection.py```) and see the controller bringing the arm's end-effector exactly at to the goal sphere.

**Comment by Erfi** [comment]
So if I make the load and goal small we can see that the goal (red sphere) is aligned with the end of the robot arm but not the "load" at the end.
![small_spheres](https://github.com/shamilmamedov/flexible_arm/assets/3267547/93537323-334b-4f04-bd3c-109441e7101e)

I think this is a URDF thing:

```
 <link name="load">
    <visual>
        <geometry>
            <sphere radius="0.01"/>
        </geometry>
        <origin rpy="0 0 0" xyz ="0.05 0 0"/>                                       **HERE: 0.05 -> 0**
    </visual>
    <inertial>
        <origin xyz="0.025 0 0" rpy="0 0 0"/>
        <mass value="0" />
        <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0" />
    </inertial>
</link>

<joint name="last_link_to_load" type="fixed">
    <parent link="virtual_link3_6"/>
    <child link="load"/>
    <axis xyz="0 0 1"/>
    <origin xyz="0.05 0 0" rpy="0 0 0"/>                                           **OR HERE: 0.05 -> 0**
</joint>
```
If I change either of these from 0.05 to 0 the problem goes away.

![small_sphere_aligned](https://github.com/shamilmamedov/flexible_arm/assets/3267547/1e5dc64c-52ca-40ee-a564-715aabb7b5e9)

@shamilmamedov  @RudolfReiter  Is this a correct solution tho? I tried to see how the End Effector position is calculated (which part of the model constitutes the end effector?) but this is not a python code: 
self.p_ee = cs.Function.load(os.path.join(model_folder, 'fkp.casadi'))

**Comment by Erfi** [comment]
And if changing the URDF is the correct solution, should I change the link or the joint? (Which makes more sense?)

**Comment by shamilmamedov** [comment]
The sphere is there to represent the load, it doesnt affect the end-effector (EE) position. We can remove the sphere as we don't actually use the load.

**Comment by shamilmamedov** [comment]
> And if changing the URDF is the correct solution, should I change the link or the joint? (Which makes more sense?)

Either remove the "load" link or change its visual parameter such that the center of sphere is aligned with the tip of the second link

**Comment by Erfi** [comment]
It looks better visually if its there, so I will just move it to the tip of the second link.

**Comment by Erfi** [comment]
Closing this issue, as it is resolved by commit e0cae4e
